### ISSUE TITLE
Is the default value (256MiB) of max_digestion_size_per_segment too larger that will consume lots of memory when merge big parts

### ISSUE THREAD

**Issue by ArthurZou** [ClickHouse/ClickHouse#67649]
### Question

here is the part of memory digraph
![img_v3_02db_609a2de3-3c97-4d2d-a748-f0138a97d29g](https://github.com/user-attachments/assets/afec92d6-c9f3-4acf-b77e-e66acf8ab246)

**Comment by cangyin** [comment]
Yes it can be, see https://github.com/ClickHouse/ClickHouse/issues/47875 .

BTW, how do you generated the memory digraph for ClickHouse ?

**Comment by ArthurZou** [comment]
using jemalloc && jeprof

**Comment by ArthurZou** [comment]
I used some random words (random character sequence ,len == 10  ) to test the performance of FstBuilder. Here is results:
| number of words       |  size of fst    |   memory consumption  while fst building  |
| --------   | ------   | ----  |
|  1w     |  340KiB   |   20MiB    |
|  10w     |  3.6MiB  |   120MiB    |
| 50w     |    19MiB    |   900 MiB |

The result shows that the number of words has significant impact to FstBuilder

So if inverted index is created on a column with mass of words, e.g., timestamp, ip, user_id, that would cause FstBuilder became big memory consumer.  256MiB original text may contain 1M rows, thus 1M words is a common case.

I wonder if is possible to provide another strategy that using number of terms instead of size of original text or both to determine when to writeSegment

**Comment by rschu1ze** [comment]
No longer relevant with the text index in 25.10+.
